packages/transaction-controller/src/utils/gas-fees.ts

```diff
@@ -12,33 +12,56 @@ import { GAS_ESTIMATE_TYPES } from '@metamask/gas-fee-controller';
 import { createModuleLogger } from '@metamask/utils';
 import { addHexPrefix } from 'ethereumjs-util';
 
+import { SWAP_TRANSACTION_TYPES } from '../constants';
 import { projectLogger } from '../logger';
-import type { TransactionParams } from '../types';
-import { UserFeeLevel, type TransactionMeta } from '../types';
+import type {
+  SavedGasFees,
+  TransactionParams,
+  TransactionMeta,
+  TransactionType,
+} from '../types';
+import { UserFeeLevel } from '../types';
 
 export type UpdateGasFeesRequest = {
   eip1559: boolean;
   ethQuery: EthQuery;
+  getSavedGasFees: () => SavedGasFees | undefined;
   getGasFeeEstimates: () => Promise<GasFeeState>;
   txMeta: TransactionMeta;
 };
 
 export type GetGasFeeRequest = UpdateGasFeesRequest & {
+  savedGasFees?: SavedGasFees;
   initialParams: TransactionParams;
   suggestedGasFees: Awaited<ReturnType<typeof getSuggestedGasFees>>;
 };
 
+/**
+ * Represents the user customizing their gas preference
+ */
+export const CUSTOM_GAS_ESTIMATE = 'custom';
+
 const log = createModuleLogger(projectLogger, 'gas-fees');
 
 export async function updateGasFees(request: UpdateGasFeesRequest) {
   const { txMeta } = request;
   const initialParams = { ...txMeta.txParams };
 
+  let savedGasFees = request.getSavedGasFees();
```

Comment:
We don't even have spend the time invoking the callback and potentially reading the state of the client preferences controllers if it's a swap type.

Could just use a ternary such as:

```
const isSwap = SWAP_TRANSACTION_TYPES.includes(txMeta.type);
const savedGasFees = isSwap ? undefined : request.getSavedGasFees();
```

Reply:
Done in https://github.com/MetaMask/core/pull/1966/commits/6be919ad602a1450800d4d451d0145fd7fb57ca8